### PR TITLE
feat(#214) Created RoomsTable with a demo index page that will only l…

### LINKED ISSUE #214: Dynamic Table Component
### What needs to be done
- Create dynamic table component that matches the branding for the website
- Table should be able to accept a 2D array of data, each inner array would represent a table row
- Table can accept html and display it as is (for actions, file upload)
- Table should be able to accept an array of headers to populate the list of headers

### Out of scope
- Table actions don't need to be functional

### PR DESCRIPTION
**Pull request by jonathanp16**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**
Please provide enough information so that others can review your pull request:

Explain the **details** for making this change. What existing problem does the pull request solve?
- This PR solves the issue of how we will structure tables as well as design them in the future. Additional table stylings were added to match the designs, and a basic filtering function was also added to the RoomTable component.
image

### Test plan (required)
Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

image
image

### Closing issues
List all issues this PR resolves.
- closes #214 

### Pre-Merge checklist 
Complete these items before requesting reviews
- [x] Static Analysis Rules Validated
- [x] All commits follow naming conventions 

### Sub-tasks Checklist
- [x] UI Implementation
- [x] UI Test
- [ ] Backend Implementation
- [ ] Backend Tests
- [ ] Integration Tests
- [ ] System Tests
- [x] Documentation

### Note
The structure of the tables was changed from a completely dynamic approach to now creating a table based on its need, and reusing them where applicable. This is mentioned in #214 and will make implementing table actions and filters much easier.

### CODE REVIEW

**Review by Alex-Hayeur — CHANGES_REQUESTED** [review]
The table is great and the code is well-commented.

I just have one question about the DemoController. Is this something you used for local testing? It looks necessary but the naming is vague.

**Comment by jonathanp16** [comment]
> The table is great and the code is well-commented.
> 
> I just have one question about the DemoController. Is this something you used for local testing? It looks necessary but the naming is vague.

It is mostly just a set of possible routes and views for others to see how the code and designs work, instead of assuming just by reading the code. I could definitely change the name but I was thinking for any component that is built that will need to be used by others, this is a good way of helping them get familiar with them

**Review by Alex-Hayeur — APPROVED** [review]
Alright, that makes sense. I'm approving this PR.

File: app/Http/Controllers/DemoController.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        return inertia('Demo/Tables/RoomTableIndex', [
+            'rooms' => Room::all(),
+        ]);
+    }
```
Comment (EvanDime) [inline app/Http/Controllers/DemoController.php:20]: these lines aren't covered by any tests. Please create a test before merging.

Comment (jonathanp16) [reply app/Http/Controllers/DemoController.php:20]: Added tests for this, resolving comment now

**Review by alecadub — APPROVED** [review]

**Review by EvanDime — APPROVED** [review]

**Pull request merged**